### PR TITLE
Смирнов -- Лабораторная работа #0 

### PR DESCRIPTION
**Pull request by AndrewSmrnv**
@kalyanovdmitry @VladimirTyagunov , please take a look

### CODE REVIEW

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        assertEquals(expectedMultiplicationResult, multiplyOperationResult);
+    }
+
+    @Test
+    public void isHashCodeDiffer() {
+
+        ComplexNumber z1 = new ComplexNumber(1, 0);
+        ComplexNumber z2 = new ComplexNumber(0, 0);
+
+        assertEquals(z1.hashCode() != z2.hashCode(), true);
```
Comment (KalyanovDO) [inline code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java]: Use assertTrue() instead.

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import org.junit.Test;
+import ru.unn.agile.ComplexNumber.model.ComplexNumber;
+import static org.junit.Assert.*;
+
+public class SmirnovAndrewTest {
+
+    @Test
+    public void isEmptyConstructorEqualsZeroConstructor() {
+
+        ComplexNumber complexNumberWithEmptyConstructor = new ComplexNumber();
```
Comment (KalyanovDO) [inline code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java:13]: Remove first empty line in each test

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package ru.unn.agile.ComplexNumber.Model;
+
+import org.junit.Test;
+import ru.unn.agile.ComplexNumber.model.ComplexNumber;
+import static org.junit.Assert.*;
```
Comment (KalyanovDO) [inline code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java]: It is unnesessary to use * in import, choose exact functions instead

**Review by KalyanovDO — CHANGES_REQUESTED** [review]

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java (hunk 1 above)
Comment (AndrewSmrnv) [reply code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java]: done

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java (hunk 2 above)
Comment (AndrewSmrnv) [reply code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java:13]: Don't think that it's a problem, just matter of style :)

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java (hunk 3 above)
Comment (AndrewSmrnv) [reply code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java]: fixed

**Comment by KalyanovDO** [comment]
:+1:

**Comment by vtyagunov** [comment]
Looks fine. 👍

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import org.junit.Test;
+import ru.unn.agile.ComplexNumber.model.ComplexNumber;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class SmirnovAndrewTest {
+
+    @Test
+    public void isEmptyConstructorEqualsZeroConstructor() {
+
```
Comment (VitKonevskiy) [inline code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java:12]: Remove empty line, pls

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    @Test
+    public void isHashCodeDiffer() {
+
+        ComplexNumber z1 = new ComplexNumber(1, 0);
+        ComplexNumber z2 = new ComplexNumber(0, 0);
+
+        assertTrue(z1.hashCode() != z2.hashCode());
+    }
+
```
Comment (VitKonevskiy) [inline code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java:49]: Remove empty line, pls

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java (hunk 4 above)
Comment (AndrewSmrnv) [reply code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java]: As I already mentioned in one of conversations with @Dmitry - blank lines in this case is ok

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java (hunk 5 above)
Comment (AndrewSmrnv) [reply code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java:49]: As I already mentioned in one of the conversations with @KalyanovDO - blank lines, in this case, is ok

File: code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java (hunk 4 above)
Comment (AndrewSmrnv) [reply code/korniakov-kirill-lab1/src/test/java/ru/unn/agile/ComplexNumber/Model/SmirnovAndrewTest.java:12]: As I already mentioned in one of the conversations with @KalyanovDO - blank lines, in this case, is ok

**Review by VitKonevskiy — APPROVED** [review]

**Pull request merged**